vsintegration/src/FSharp.Editor/LanguageService.fs

```diff
@@ -74,35 +75,63 @@ type internal FSharpLanguageService(package : FSharpPackage) =
         let result = VsRunningDocumentTable.FindDocumentWithoutLocking(package.RunningDocumentTable,filename)
         match result with
         | Some (hier, _) ->
-            match hier with
-            | :? IProvideProjectSite as siteProvider ->
-                let site = siteProvider.GetProjectSite()
-                let projectGuid = Guid(site.ProjectGuid)
-                let projectFileName = site.ProjectFileName()
-                let projectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFileName, projectFileName)
-
-                let options = ProjectSitesAndFiles.GetProjectOptionsForProjectSite(site, site.ProjectFileName())
-                if not (optionsCache.ContainsKey(projectId)) then
-                    optionsCache.Add(projectId, options)
-
-                if obj.ReferenceEquals(workspace.ProjectTracker.GetProject(projectId), null) then
-                    let projectContextFactory = this.Package.ComponentModel.GetService<IWorkspaceProjectContextFactory>();
-                    let errorReporter = ProjectExternalErrorReporter(projectId, "FS", this.SystemServiceProvider)
-                    let outputFlag = site.CompilerFlags() |> Seq.pick(fun flag ->
-                        if flag.StartsWith("-o:") then Some(flag.Substring(3))
-                        else if flag.StartsWith("--out:") then Some(flag.Substring(6))
-                        else None)
-                    let outputPath = if Path.IsPathRooted(outputFlag) then outputFlag else Path.Combine(Path.GetDirectoryName(projectFileName), outputFlag)
-
-                    let projectContext = projectContextFactory.CreateProjectContext(FSharpCommonConstants.FSharpLanguageName, projectFileName, projectFileName, projectGuid, hier, outputPath, errorReporter)
-                    let project = projectContext :?> AbstractProject
-
-                    this.SyncProject(project, projectContext, site)
-                    site.AdviseProjectSiteChanges(FSharpCommonConstants.FSharpLanguageServiceCallbackName, AdviseProjectSiteChanges(fun () -> this.SyncProject(project, projectContext, site)))
-                    site.AdviseProjectSiteClosed(FSharpCommonConstants.FSharpLanguageServiceCallbackName, AdviseProjectSiteChanges(fun () -> project.Disconnect()))
-            | _ -> ()
+            if IsScript(filename) then
+                this.SetupStandAloneFile(filename, workspace, hier)
+            else
+                match hier with
+                | :? IProvideProjectSite as siteProvider -> this.SetupProjectFile(siteProvider, workspace)
+                | _ -> ()
         | _ -> ()
 
+    member this.SetupProjectFile(siteProvider: IProvideProjectSite, workspace: VisualStudioWorkspaceImpl) =
+        let site = siteProvider.GetProjectSite()
+        let projectGuid = Guid(site.ProjectGuid)
+        let projectFileName = site.ProjectFileName()
```

Comment:
Should this be handling exceptions from reading files on disk?

Should it get the text from the in-editor file, since it's in the RDT now.